tests/test_filesystem.py

```diff
@@ -1,138 +1,878 @@
+import logging
 import os
+import re
+import shutil
 import tempfile
 import unittest
 from pathlib import Path
 from unittest.mock import patch
 
 import pytest
 import yaml
+from dotenv import load_dotenv
+from magpie.models import Permission, Route
+from magpie.permissions import Access, Scope
+from magpie.services import ServiceAPI
+from webtest.app import TestApp
 
-from cowbird.handlers.impl.filesystem import NOTEBOOKS_DIR_NAME
-from tests import utils
+from cowbird.api.schemas import ValidOperations
+from cowbird.handlers import HandlerFactory
+from cowbird.handlers.impl.filesystem import NOTEBOOKS_DIR_NAME, SECURE_DATA_PROXY_NAME
+from cowbird.typedefs import JSON
+from tests import test_magpie, utils
+
+CURR_DIR = Path(__file__).resolve().parent
 
 
 @pytest.mark.filesystem
 class TestFileSystem(unittest.TestCase):
     """
-    Test FileSystem operations.
+    Test FileSystem parent class, containing some utility functions and common setup/teardown operations.
     """
```

Comment:
Doesn't this get picked up by pytest/unittest as a separate test class (that is executed on its own) even though there are no test cases in it (because of the `Test` prefix). 

If so, consider adding a `Base`, `Util`, or some other prefix. 
This will also highlight the fact that it is intended to be extended by other test classes, and only provides utilities instead of being test cases on their own.

Reply:
If I run the tests, I don't necessarily see tests from this class, as it doesn't contain any.

I think I could add a `Base` prefix to make it more clear that it doesn't contain any tests, but I think unittest still detects it as a test class since it inherits `unittest.TestCase`. I need this base class to inherit `unittest.TestCase` though since I want this class to execute the setup and teardown functions during the tests of the children classes.

Reply:
Ok. As long as it doesn't produce a +1 test entry, that's fine. Even if it did, not a deal-breaker either.